diff --git a/build.rs b/build.rs
index 3199616..5e6777b 100644
--- a/build.rs
+++ b/build.rs
@@ -9,38 +9,12 @@ use std::{
 };
 
 fn main() {
-    if std::env::var("DOCS_RS").is_ok() {
-        // don't build c library on docs.rs due to network restrictions
-        return;
-    }
-
-    // Select commit based on features. If conflict, always choose newest release
-    let tag = if cfg!(feature = "v3-1-1") {
-        "refs/tags/rel-3.1.1"
-    } else if cfg!(feature = "v3-1-0") {
-        "refs/tags/rel-3.1.0"
-    } else if cfg!(feature = "v3-0-0") {
-        "refs/tags/rel-3.0.0"
-    } else if cfg!(feature = "sc2022-light") {
-        "refs/tags/sc2022-light"
-    } else if cfg!(feature = "sc2022-hyper") {
-        "refs/tags/sc2022-hyper"
-    } else if cfg!(feature = "sc2022-bulky") {
-        "refs/tags/sc2022-bulky"
-    } else {
-        // default to newest version
-        "refs/tags/rel-3.1.1"
-    };
-
-    // Build C library
-    // Full commit hash needs to be provided
-    build("https://github.com/arminbiere/kissat.git", "master", tag);
-
-    let out_dir = env::var("OUT_DIR").unwrap();
+    let out_dir = env::var("NIX_KISSAT_DIR").unwrap();
 
     // Built solver is in out_dir
     println!("cargo:rustc-link-search={}", out_dir);
     println!("cargo:rustc-link-search={}/lib", out_dir);
+    println!("cargo:rustc-link-lib=kissat");
 }
 
 fn build(repo: &str, branch: &str, reference: &str) {
